src/renderer/mod.rs

```diff
@@ -81,11 +81,62 @@ impl<'a> Renderer<'a> {
             return Ok(to_value(to_string_pretty(context).unwrap()).unwrap());
         }
 
+        #[inline]
+        fn evaluate_sub_variable(context: &Value, key: &str, tpl_name: &str) -> String {
+            let sub_vars_to_calc = pull_out_square_bracket(key);
+            let mut new_key = key.to_string();
+
+            for sub_var in sub_vars_to_calc.iter() {
+                // Translate from variable name to variable value
+                let post_var = find_variable(context, sub_var.as_ref(), tpl_name).unwrap();
+                let post_var_as_str = {
+                    if post_var.is_string() {
+                        post_var.as_str().unwrap().to_string()
+                    } else if post_var.is_f64() {
+                        post_var.as_f64().unwrap().to_string()
+                    } else if post_var.is_i64() {
+                        post_var.as_i64().unwrap().to_string()
+                    } else if post_var.is_boolean() {
+                        post_var.as_bool().unwrap().to_string()
+                    } else {
+                        panic!("Unknown Type")
+                    }
+                };
+
+                // Rebuild the original key String replacing variable name with value
+                let nk = new_key.clone();
+                let divider = "[".to_string() + sub_var + "]";
+                let mut the_parts = nk.splitn(2, divider.as_str());
+
+                new_key = String::from(
+                    the_parts.next().unwrap().to_string()
+                        + "."
+                        + post_var_as_str.as_ref()
+                        + the_parts.next().unwrap_or(""),
+                );
+            }
+            new_key
+                .replace("['", ".")
+                .replace("[\"", ".")
+                .replace("[", ".")
+                .replace("']", "")
+                .replace("\"]", "")
+                .replace("]", "")
+        }
+
         #[inline]
         fn find_variable(context: &Value, key: &str, tpl_name: &str) -> Result<Value> {
-            match context.pointer(&get_json_pointer(key)) {
+            let key_s = match key.contains('[') {
+                false => key.into(),
+                true => evaluate_sub_variable(context, key, tpl_name),
+            };
```

Comment:
clippy will probably tell you to use `if else` rather than matching on true/false, a good idea here imo

Reply:
I actually prefer this style. Why is if/else better?

But clippy told me to change it. And we should obey clippy so I'll change it.